lib/syscall_shim/uk_syscall_binary.c

```diff
@@ -38,11 +38,17 @@
 #include <uk/arch/ctx.h>
 #include <uk/assert.h>
 #include <uk/essentials.h>
+#include <uk/thread.h>
 #include "arch/regmap_linuxabi.h"
 #if CONFIG_LIBSYSCALL_SHIM_STRACE
 #include <uk/plat/console.h> /* ukplat_coutk */
 #endif /* CONFIG_LIBSYSCALL_SHIM_STRACE */
 
+struct uk_syscall_ctx {
+	struct ukarch_execenv execenv;
+	__uptr auxsp;
```

Comment:
On syscall C handler entry we are no longer given guarantee that registers have been switched to those of Unikraft's, so therefore we have no direct access to `auxsp`. However we did have it for a brief time on `swapgs` which is when we also push this. I will thoroughly comment this as it's basically an optimization to avoid having to do any `swapgs`/`rdgsbase`/etc instructions more than needed as they cost lots of CPU cycles.